app/controllers/process_plates_controller.rb

```diff
@@ -36,12 +36,12 @@ def create
           call_external_services(barcodes) if barcodes && receive_plates_process
 
           # add a flash on the page for the number of unique barcodes scanned in
-          num_unique_barcodes = bed_layout_verification.process_plate&.barcodes.uniq.length
-          if num_unique_barcodes
-            flash[:notice] = "Success - #{num_unique_barcodes} unique plate(s) scanned"
-          else
-            flash[:notice] = "Success"
-          end
+          num_unique_barcodes = bed_layout_verification.process_plate&.num_unique_barcodes
```

Comment:
The safe navigation operator made no sense here, as it would just result in .uniq being called on nil, which would blow up in turn. Rather than the messy ```bed_layout_verification.process_plate&.barcodes&.uniq&.length``` I opted to move the barcode counting into ProcessPlate.

I don't think this code will even see a nil process_plate (legacy processes don't se it, but error before here anyway) but keeping changes minimal for now to reduce clashes on merging other stuff.